plugin/fireplace.vim

```diff
@@ -1666,6 +1666,42 @@ augroup fireplace_go_to_file
   autocmd FileType clojure call s:set_up_go_to_file()
 augroup END
 
+" Section: Spec
+" TODO: `spec-list`
+
+function! fireplace#spec(op, symbol) abort
+  if fireplace#op_available(a:op)
+    let unaliased_symbol = fireplace#eval(a:symbol)
+    let response = fireplace#message({'op': a:op, 'spec-name': unaliased_symbol})[0]
+    if !empty(get(response, a:op))
+      echo get(response, a:op)
```

Comment:
For `spec-form`, this is just dumping a raw Vim data structure. Surely we can do better?

Reply:
The "SpecExample" result is sent through the pretty printer in orchard/cider. I'll separate the two functions and see how i can make the SpecForm result more clojury

Reply:
If this was a rethorical question, i need some hints how to deal with the output

Reply:
I assume what's needed is some sort of recursive function that turns `['a', ['b', 'c']]' into `(a (b c))`.

Reply:
Something like this perhaps:

```vim
function! s:pr(obj) abort
  if type(a:obj) == v:t_string
    return a:obj
  elseif type(a:obj) == v:t_list
    return '(' . join(map(copy(a:obj), 's:pr(v:val)'), ' ') . ')'
  elseif type(a:obj) == v:t_dict
    return '{' . join(map(keys(a:obj), 's:pr(v:val) . " " . s:pr(a:obj[v:val])'), ', ') . '}'
  else
    return string(a:obj)
  endif
endfunction
```

It does the right thing for plain strings too so no need to special case.